db/migrate/20191215115843_create_device_story_devices.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class CreateDeviceStoryDevices < ActiveRecord::Migration
+  def change
+    create_table :device_story_devices do |t|
+      t.references :user
+
+      t.string :device_urn, null: false, default: ''
+      t.jsonb :metadata, null: false, default: '{}'
```

Comment:
I understand why these are marked `NOT NULL`. However, I don't think that they should have defaults. We would never want to have a table row where the device URN is the empty string, or the metadata is completely empty. Those are error states, not valid data.

Reply:
I see. I agree that we should not set default values. I removed `default` clauses. For `metadata` column, I removed `null` constraint as well. I thought that there could be a case that we would want to create device page before setting it up. In that case, we would not have metadata because device would not be working yet.